### PR TITLE
Bump `nix` to 0.22

### PR DESCRIPTION
**Pull request by Aaron1011**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -151,8 +151,5 @@ mod loop_logic;
 mod sources;
 
 fn no_nix_err(err: nix::Error) -> std::io::Error {
-    match err {
-        ::nix::Error::Sys(errno) => errno.into(),
-        _ => unreachable!(),
-    }
+    err.into()
 }
```
Comment (vberger) [inline src/lib.rs:155]: If there is now an `Into<std::io::Error>` impl for `nix::Errno`, then this `no_nix_err` function should just be removed, as it no longer serves any purpose.

Comment (Aaron1011) [reply src/lib.rs:155]: Keeping this function helps with type inference in several places (where we call `.map_err(no_nix_err)`)

Comment (vberger) [reply src/lib.rs:155]: I'd prefer if those places where properly fixed rather than keeping `no_nix_err` as a mere shim around `Into::into()`.

**Comment by vberger** [comment]
Done in #59.

**Pull request closed without merging**